packages/mui-material/src/Button/Button.js

```diff
@@ -295,9 +296,23 @@ const ButtonEndIcon = styled('span', {
   ...commonIconStyles(ownerState),
 }));
 
+const addPositionDataAttr = (buttonGroupButtonContext) => {
+  if (buttonGroupButtonContext.firstButton && buttonGroupButtonContext.lastButton) {
+    return {};
```

Comment:
Why do we need this special case? Wouldn't having both `data-first-child` and `data-last-child` suffice?

Reply:
Because if first child is found for a single button inside the button group, the `data-first-child` attribute is _returned_:
```js
  if (buttonGroupButtonContext.firstButton) {
    return { 'data-first-child': '' };
  }
```
and `data-last-child` is not applied.

If you mean by adding both like:
```js
  if (buttonGroupButtonContext.firstButton && buttonGroupButtonContext.lastButton) {
    return { 'data-first-child': '', 'data-last-child': '' };
  }
```
then the styles won't be correct for a single button.